src/realm-settings/ProfilesTab.tsx

```diff
@@ -38,31 +40,33 @@ export const ProfilesTab = () => {
   const [show, setShow] = useState(false);
   const [key, setKey] = useState(0);
 
-  const loader = async () => {
-    const allProfiles = await adminClient.clientPolicies.listProfiles({
-      realm,
-      includeGlobalProfiles: true,
-    });
+  useFetch(
+    () =>
+      adminClient.clientPolicies.listProfiles({
+        includeGlobalProfiles: true,
+      }),
+    (allProfiles) => {
+      setGlobalProfiles(allProfiles.globalProfiles);
 
-    setGlobalProfiles(allProfiles.globalProfiles);
-
-    const globalProfiles = allProfiles.globalProfiles?.map(
-      (globalProfiles) => ({
-        ...globalProfiles,
-        global: true,
-      })
-    );
+      const globalProfiles = allProfiles.globalProfiles?.map(
+        (globalProfiles) => ({
+          ...globalProfiles,
+          global: true,
+        })
+      );
 
-    const profiles = allProfiles.profiles?.map((profiles) => ({
-      ...profiles,
-      global: false,
-    }));
+      const profiles = allProfiles.profiles?.map((profiles) => ({
+        ...profiles,
+        global: false,
+      }));
 
-    const allClientProfiles = globalProfiles?.concat(profiles ?? []);
-    setTableProfiles(allClientProfiles);
+      const allClientProfiles = globalProfiles?.concat(profiles ?? []);
+      setTableProfiles(allClientProfiles || []);
+    },
+    [key]
+  );
 
-    return allClientProfiles ?? [];
-  };
+  const loader = async () => Promise.resolve(tableProfiles || []);
```

Comment:
Same here, this is an async function you should be able to return the value directly.
```suggestion
  const loader = async () => tableProfiles ?? [];
```